### PR TITLE
Add param to include t-h-t environment files

### PR DESCRIPTION
**Pull request by olliewalsh**
Add HeatEnv param to include built-in t-h-t environment files in stack
create command line.

### CODE REVIEW

File: api/v1beta1/openstackconfiggenerator_types.go [hunk 1]
```diff
@@ -24,6 +24,8 @@ import (
 type OpenStackConfigGeneratorSpec struct {
 	// Name of the image used to generate configs. If missing will be set to the configured OPENSTACKCLIENT_IMAGE_URL_DEFAULT in the CSV for the OSP Director Operator.
 	ImageURL string `json:"imageURL,omitempty"`
+	// Optional. List of heat environment files to include from tripleo-heat-templates/environments
+	HeatEnvs []string `json:"heatEnvs,omitempty"`
```
Comment (stuggi) [inline api/v1beta1/openstackconfiggenerator_types.go:28]: maybe default to an empty list like here https://github.com/openstack-k8s-operators/osp-director-operator/blob/master/api/v1beta1/openstacknetattachment_types.go#L37

File: controllers/openstackconfiggenerator.go [hunk 2]
```diff
@@ -263,6 +264,18 @@ func (r *OpenStackConfigGeneratorReconciler) Reconcile(ctx context.Context, req
 		return ctrl.Result{}, err
 	}
 
+	// add build-in heat environment files
+	var tripleoEnvironmentFiles []string
+	if instance.Spec.HeatEnvs != nil {
```
Comment (stuggi) [inline controllers/openstackconfiggenerator.go:269]: with the above set default to an empty slice I think we won't require this.

File: config/crd/bases/osp-director.openstack.org_openstackconfiggenerators.yaml [hunk 3]
```diff
@@ -89,6 +89,11 @@ spec:
                 description: Required. the name of the config map containing Heat
                   env file customizations
                 type: string
+              heatEnvs:
+                description: List of heat environment files to include from tripleo-heat-templates/environments
```
Comment (stuggi) [inline config/crd/bases/osp-director.openstack.org_openstackconfiggenerators.yaml:93]: should we be here really specific that it is just the file name without any path?

File: controllers/openstackconfiggenerator.go (hunk 2 above)
Comment (olliewalsh) [reply controllers/openstackconfiggenerator.go:269]: yes, avoiding the nil check is why I wanted to set a default

File: api/v1beta1/openstackconfiggenerator_types.go (hunk 1 above)
Comment (olliewalsh) [reply api/v1beta1/openstackconfiggenerator_types.go:28]: will try it now...

File: config/crd/bases/osp-director.openstack.org_openstackconfiggenerators.yaml (hunk 3 above)
Comment (olliewalsh) [reply config/crd/bases/osp-director.openstack.org_openstackconfiggenerators.yaml:93]: can include a path, but it's relative to t-h-t/environments

Comment (olliewalsh) [reply config/crd/bases/osp-director.openstack.org_openstackconfiggenerators.yaml:93]: e.g ssl/enable-internal-tls.j2.yaml

Comment (stuggi) [reply config/crd/bases/osp-director.openstack.org_openstackconfiggenerators.yaml:93]: oh right, sure. yes we need subnet references.

File: api/v1beta1/openstackconfiggenerator_types.go [hunk 4]
```diff
@@ -24,6 +24,10 @@ import (
 type OpenStackConfigGeneratorSpec struct {
 	// Name of the image used to generate configs. If missing will be set to the configured OPENSTACKCLIENT_IMAGE_URL_DEFAULT in the CSV for the OSP Director Operator.
 	ImageURL string `json:"imageURL,omitempty"`
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default={}
```
Comment (stuggi) [inline api/v1beta1/openstackconfiggenerator_types.go:28]: well I am not sure if that really works as I was expecting it to do. we do not see a default in the rendered crd yaml. maybe we should default it in the webhook if needed.

Comment (olliewalsh) [reply api/v1beta1/openstackconfiggenerator_types.go:28]: Does the right thing AFAICT. I deployed with and without this set

Comment (olliewalsh) [reply api/v1beta1/openstackconfiggenerator_types.go:28]: No default in the crd yaml when it's optional maybe?

**Review by dprince — APPROVED** [review]

**Pull request merged**